torch/_inductor/runtime/triton_helpers.py

```diff
@@ -16,15 +16,21 @@ class tl:  # type: ignore[no-redef]
 
 # In the latest triton, math functions were shuffled around into different modules:
 # https://github.com/openai/triton/pull/3172
-if hasattr(tl.extra, "cuda") and hasattr(tl.extra.cuda, "libdevice"):
-    libdevice = tl.extra.cuda.libdevice
-    math = tl.math
-elif hasattr(tl.extra, "intel") and hasattr(tl.extra.intel, "libdevice"):
-    libdevice = tl.extra.intel.libdevice
+try:
+    from triton.language.extra import libdevice
+
+    libdevice = tl.extra.libdevice  # noqa: F811
     math = tl.math
-else:
-    libdevice = tl.math
-    math = tl
+except ImportError:
```

Comment:
NIT: can you use `hasattr(tl.extra, "libdevice")` to be more consistent with the existing code.

Reply:
I don't think so - it looks like `libdevice` is not part of the `__init__.py` for the `tl.extra` package. The import check is basically checking to see if the `libdevice.py` file is there.